spec/services/avatar_generator_spec.rb

```diff
@@ -1,13 +1,12 @@
 require 'rails_helper'
 
 describe AvatarGenerator do
-  let(:auth_token) { create(:auth_token) }
-
   subject do
-    described_class.new(auth_token.value)
+    described_class.new('test_identifier')
   end
 
   it 'creates proper url' do
-    expect(URI.parse(subject.generate_avatar_url)).to be_a URI::HTTPS
+    expect(URI.parse(subject.generate_url)).to be_a URI::HTTPS
+    expect(subject.generate_url).to include('test_identifier')
   end
```

Comment:
Włożyłbym to jeszcze (dla przejrzystości kodu) w dodatkowy kontekst testowania metody. No i wydaje mi się, że powinniśmy testować cały url - test taki jaki mamy teraz nie gwarantuje zachowania, które chcemy osiągnąć:

`describe '#generate_url' do`
..`subject do`
....`described_class.new('test_identifier').generate_url`
..`end`
..
..`it 'creates proper url' do`
....`expect(subject).to eq 'https://api.adorable.io/avatars/80/test_identifier'`
..`end`
`end`

Po tej zmianie co prawda mamy zahardcodowane w teście, że używamy `api.adorable.io`, ale tylko tymczasowo, tj. dopóki nie będziemy potrzebować funkcjonalności podmiany tego API. W myśl TDD, że piszemy najprostszy i najmniejszy kawałek testu, który zfailuje, jest to bardziej poprawne - bo mamy tylko (i aż) tyle testów i funkcjonalności ile potrzebujemy.